### PR TITLE
Implementing Service Discovery in Kubescape Cloud Operator

### PR DESCRIPTION
**Pull request by amirmalka**
## PR Type:
Enhancement

___
## PR Description:
This PR introduces dynamic service discovery in the Kubescape Cloud Operator. It replaces hardcoded URLs with dynamic fetching from a server, enhancing the system's flexibility to adapt to changes in the server's URLs without requiring code changes. The PR also includes updates to various Kubernetes templates to accommodate this new feature. The hardcoded URLs and certain environment variables have been removed, and a new job for service discovery has been added that fetches services from the server and updates the configmap.

___
## PR Main Files Walkthrough:
`charts/kubescape-cloud-operator/README.md`: Removed the URL override options from the Kubescape configuration.
`charts/kubescape-cloud-operator/assets/otel-collector-config.yaml`: Updated the exporter configuration to use the environment variable for the endpoint.
`charts/kubescape-cloud-operator/templates/configs/cloudapi-configmap.yaml`: Added new fields for metrics and services, and removed hard-coded URLs.
`charts/kubescape-cloud-operator/templates/kubescape/configmap.yaml`: Removed the cloud URL fields from the Kubescape configuration.
`charts/kubescape-cloud-operator/templates/servicediscovery/job.yaml`: Added a new job for service discovery that fetches services from the server and updates the configmap.
`charts/kubescape-cloud-operator/templates/servicediscovery/role.yaml`: Added a new role for service discovery with permissions to update and create configmaps.
`charts/kubescape-cloud-operator/templates/servicediscovery/rolebinding.yaml`: Added a new role binding for the service discovery role.
`charts/kubescape-cloud-operator/values.yaml`: Removed the hardcoded URLs and added a new field for the server.
`charts/kubescape-cloud-operator/templates/kubescape/deployment.yaml`: Removed the environment variable for SaaS environment and added a new volume mount for services.json if the server is defined.
`charts/kubescape-cloud-operator/templates/gateway/deployment.yaml`: Added a new volume mount for services.json if the server is defined.

### CODE REVIEW

**Comment by amirmalka** [comment]
TODO: 
- Revert image tags before committing changes

File: charts/kubescape-cloud-operator/templates/servicediscovery/serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{{- if .Values.server  }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: service-discovery
```
Comment (matthyx) [inline charts/kubescape-cloud-operator/templates/servicediscovery/serviceaccount.yaml:5]: I think this name I would put in values.yaml like for the other components

**Review by matthyx — COMMENTED** [review]
looks almost perfect, let's modify the job and wait for images to be released

**Comment by amirmalka** [comment]
Ready to be merged:
https://github.com/kubescape/helm-charts/actions/runs/6021436772

File: charts/kubescape-cloud-operator/templates/otel-collector/deployment.yaml [hunk 2]
```diff
@@ -54,6 +54,11 @@ spec:
             containerPort: 4317
             protocol: TCP
         env:
+        - name: CLOUD_OTEL_COLLECTOR_URL
+          valueFrom: 
+            configMapKeyRef:
+              name: {{ .Values.global.cloudConfig }}
```
Comment (matthyx) [inline charts/kubescape-cloud-operator/templates/otel-collector/deployment.yaml:60]: nice !

**Pull request merged**